client/allocrunner/taskrunner/service_hook.go

```diff
@@ -179,6 +182,17 @@ func (h *serviceHook) deregister() {
 
 }
 
+func (h *serviceHook) Stop(ctx context.Context, req *interfaces.TaskStopRequest, resp *interfaces.TaskStopResponse) error {
+	taskServices := h.getTaskServices()
```

Comment:
Should this call `h.deregister()` or hold the lock like `Exited` and `PreKilling`?

Reply:
fixed, yeah